woocommerce/woocommerce-ios
# [UL&S] When logging with site address, redirect users to enter the wordpress.com email address

Closes #3426 

Before:

https://user-images.githubusercontent.com/2722505/103692828-59fddd00-4f66-11eb-9322-d6a23c29f415.mov

After:

https://user-images.githubusercontent.com/2722505/103692843-61bd8180-4f66-11eb-8443-73baa1740695.mov

As usual, most of the solution is implemented in WPAuthenticator. The corresponding PR is wordpress-mobile/WordPressAuthenticator-iOS#555

## Changes
* Point WPAuthenticator to the branch that implements the logic to navigate to the expected view
* Update the logic in AuthenticationManager

## How to test
* Attempt to login with a self-hosted site (i.e. a site created with jurassic ninja)
* Make sure the new site is connected to a valid Jetpack account
* Checkout the branch, run `bundle exec pod install`
* Log out if necessary. 
* Select "Enter your Store Address". Enter the address of the new site.
* Notice if the authentication flow requests an email account. 
* Continue the login process with the email linked to the Jetpack account connected to the site.

Update release notes:
